### PR TITLE
Cherry-pick: Fix basic auth (#12367)

### PR DESCRIPTION
**Pull request by dbadura**
* fix basic auth

* add tests

* fix imports

* fix after review

**Description**

Changes proposed in this pull request:

- cherry-pick git auth fix

**Related issue(s)**

Bump: https://github.com/kyma-project/kyma/pull/12497

### CODE REVIEW

**Review by pPrecel — APPROVED** [review]
LGTM

File: components/function-controller/cmd/jobinit/main.go [hunk 1]
```diff
@@ -34,7 +34,11 @@ func main() {
 	log.Printf("Clone repo from url: %s and commit: %s...\n", cfg.RepositoryUrl, cfg.RepositoryCommit)
 	commit, err := operator.Clone(cfg.MountPath, gitOptions)
 	if err != nil {
-		log.Fatalln(errors.Wrapf(err, "while cloning repository: %s, from commit: %s", cfg.RepositoryUrl, cfg.RepositoryCommit))
+		if git.IsAuthErr(err) {
+			log.Printf("while cloning repository bad credentials were provided, errMsg: %s", err.Error())
```
Comment (pPrecel) [inline components/function-controller/cmd/jobinit/main.go:38]: suggestion only 😄 
alt text

```suggestion
			log.Printf("bad credentials provided while cloning the repository, errMsg: %s", err.Error())
```

**Review by pPrecel — COMMENTED** [review]
Should we create PR with all those changes to the `main`? Or is this something only for the `release-1.24` branch?

**Comment by dbadura** [comment]
Those changes are coming from main branch

**Pull request merged**